thavalon-webapp/src/utils/GameSocket.tsx

```diff
@@ -1,70 +1,150 @@
  * Will contain a websocket, that can either be retrieved or optionally made.
  */
 export class GameSocket {
     // the underlying gamesocket instance
-    private static instance: GameSocket;
+    private static instance: GameSocket | undefined;
     // the instantiated instance of the underlying websocket
     // can also be undefined since it's not explicitly set in constructor
     private websocket: WebSocket;
+    // event handler for lobby events
+    private _onLobbyEvent: SimpleEventDispatcher<InboundMessage>;
 
-    // create the underlying websocket in the constructor
+    /**
+     * Construct the underlying websocket instance and set up function handlers.
+     * @param socketUrl The socketUrl for the websocket
+     */
     private constructor(socketUrl: string) {
         this.websocket = new WebSocket(socketUrl);
+        this._onLobbyEvent = new SimpleEventDispatcher<InboundMessage>();
         this.websocket.onopen = this.socketOnOpen.bind(this);
         this.websocket.onmessage = this.socketOnMessage.bind(this);
         this.websocket.onclose = this.socketOnClose.bind(this);
         this.websocket.onerror = this.socketOnError.bind(this);
-    } 
-    // functions for handling incoming websocket events
+    }
+
+    /**
+     * Listen to websocket's onopen event.
+     * @param event Event received when socket open.
+     */
     private socketOnOpen(event: Event) {
         console.log("Successfully initiated connection.");
     }
 
+    /**
+     * Listen to websocket's onmessage event.
+     * @param event Event received when socket gets message.
+     */
     private socketOnMessage(event: MessageEvent) {
         console.log(event);
         console.log("Received message: " + event.data);
+        const message = JSON.parse(event.data);
+        this._onLobbyEvent.dispatch(message);
     }
 
+    /**
+     * Listen to websocket's onclose event.
+     * @param event Event received when socket is closed.
+     */
     private socketOnClose(event: CloseEvent) {
         console.log("Recieved on close message.");
     }
 
+    /**
+     * Listen to websocket's onerror event.
+     * @param event Event received when socket errors.
+     */
     private socketOnError(event: Event) {
         console.log("Received on error message.");
     }
 
+    /**
+     * Send a message on the websocket. This will wait until websocket is
+     * open before sending the message.
+     * @param outboundMessage The outboundMessage to be sent to the server. 
+     */
+    public sendMessage(outboundMessage: OutboundMessage) {
+        if (this.websocket.readyState === WebSocket.OPEN) {
+            console.log("Sending message of type: " + outboundMessage.messageType);
+            this.websocket.send(JSON.stringify(outboundMessage));
+            return;
+        }
+        // check again if websocket ready in 10 milliseconds
+        setTimeout(() => this.sendMessage(outboundMessage), 1000);

```

Comment:
this was bad in my PR, that should be 10 not 1000

Reply:
Fixed